hedera-node/src/main/java/com/hedera/services/legacy/services/state/AwareProcessLogic.java

```diff
@@ -144,7 +158,11 @@ private boolean txnSanityChecks(PlatformTxnAccessor accessor, Instant consensusT
 	}
 
 	private void processTxnInCtx() {
-		doProcess(ctx.txnCtx().accessor(), ctx.txnCtx().consensusTime());
+		doProcess((PlatformTxnAccessor)ctx.txnCtx().accessor(), ctx.txnCtx().consensusTime());
```

Comment:
Maybe we could check whether the `accessor` is of type `PlatformTxnAccessor` and if it is not -> throw with illegal argument exception?  Same thing for `processTriggeredTxnInCtx`